addons/web/static/src/js/widgets/user_menu.js

```diff
@@ -26,8 +26,24 @@ var SystrayMenu = Widget.extend({
                 f($(this));
             }
         });
+
         this.$el.parent().show();
-        return this._super.apply(this, arguments);
+        this._super.apply(this, arguments);
+
+        return self.alive(new Model('res.users').call('read', [[session.uid], ['tz_offset']])).then(function(result) {
+            var user_offset = result[0]['tz_offset'];
+            var offset = -(new Date().getTimezoneOffset());
+            var browser_offset = (offset < 0) ? "-" : "+";
+            browser_offset += _.str.sprintf("%02d", Math.abs(offset / 60));
+            browser_offset += _.str.sprintf("%02d", Math.abs(offset % 60));
+            if (browser_offset !== user_offset) {
+                $('.oe_timezone_icon').append('<i class="fa fa-exclamation-triangle" style="color:red;" title="Your Odoo preference timezone does not match your browser timezone: \n Your Odoo timezone is '+user_offset+' \n Your Browser timezone is '+browser_offset+'"></i>');
```

Comment:
Why not creating seperate template may be with name WebClient.timezone_warning, it will be more readable